source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionCanBeCancelledWhenRetriesAreEnabled.cs

```diff
@@ -377,6 +380,72 @@ public async Task DuringStartScript_ForListeningTentacle_ThatIsRetryingTheRpc_An
                     if (scriptServiceRecordedUsages.ForStartScriptAsync().Started >= 2)
                     {
                         cancelExecutionCancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(2));
+                        Logger.Information("Cancelling cancellation token source after 2 seconds.");
+                        return;
+                    }
+
+                    await Task.Delay(TimeSpan.FromSeconds(0.5), CancellationToken);
+                }
+            }, CancellationToken);
+
+            var expectedException = new ExceptionContractAssertionBuilder(FailureScenario.ScriptExecutionCancelled, tentacleConfigurationTestCase.TentacleType, clientAndTentacle).Build();
+            await AssertionExtensions.Should(async () => await executeScriptTask).ThrowExceptionContractAsync(expectedException);
+            await cancellationTask;
+
+            // Assert
+            scriptServiceRecordedUsages.ForStartScriptAsync().Completed.Should().BeGreaterOrEqualTo(2);
+            scriptServiceRecordedUsages.ForGetStatusAsync().Started.Should().Be(0);
+            scriptServiceRecordedUsages.ForCancelScriptAsync().Started.Should().BeGreaterOrEqualTo(0, "Script Execution does not need to be cancelled on Tentacle as it has not started");
+            scriptServiceRecordedUsages.ForCompleteScriptAsync().Started.Should().Be(0);
+        }
+
+        [Test]
+        [TentacleConfigurations(testPolling: false)]
+        public async Task DuringStartScript_ForListeningTentacle_ThatIsRetryingTheRpc_AndConnectingWithPooledConnection_ScriptExecutionCanBeCancelled(TentacleConfigurationTestCase tentacleConfigurationTestCase)
```

Comment:
This is the same test as before, but causing it to fail on a connection that was in the pool.

I.e., it's an exact copy except to make it hit the pool:
- Leave `TcpClientPooledConnectionTimeout` as it's 9 minute default (so we don't change it)
- No longer `Delay` after we close the connection

Reply:
This test is flaky, attempting to drop it in this PR https://github.com/OctopusDeploy/OctopusTentacle/pull/950